src/Response/ResponseExtensions.cs

```diff
@@ -58,5 +63,118 @@ public static class ResponseExtensions
+                rangeEnd = contentLength - 1;
+                await GetResponseBodyDelegate(rangeStart, rangeEnd, contentLength, source, response.Body, response.HttpContext.RequestAborted);
+            }
+        }
+
+        private static async Task GetResponseBodyDelegate(long rangeStart, long rangeEnd, long contentLength, Stream source, Stream destination, CancellationToken cancellationToken)
+        {
+            if (rangeStart == 0 && rangeEnd == contentLength - 1)
+            {
+                await StreamCopyOperation.CopyToAsync(source, destination, new long?(), 65536, cancellationToken);
+            }
+            else
+            {
+                if (!source.CanSeek)
+                {
+                    throw new InvalidOperationException("Sending Range Responses requires a seekable stream eg. FileStream or MemoryStream");
+                }
+
+                var totalBytesToSend = rangeEnd - rangeStart + 1;
+                const int BufferSize = 0x1000;
+                var buffer = new byte[BufferSize];
+                var bytesRemaining = totalBytesToSend;
+
+                source.Seek(rangeStart, SeekOrigin.Begin);
+                while (bytesRemaining > 0)
+                {
+                    var count = bytesRemaining <= buffer.Length
+                        ? await source.ReadAsync(buffer, 0, (int)Math.Min(bytesRemaining, int.MaxValue), cancellationToken)
+                        : await source.ReadAsync(buffer, 0, buffer.Length, cancellationToken);
+
+                    try
+                    {
+                        await destination.WriteAsync(buffer, 0, count, cancellationToken);
+                        await destination.FlushAsync(cancellationToken);
+                        bytesRemaining -= count;
+                    }
+                    catch (Exception httpException)
+                    {
+                        /* in Asp.Net we can call HttpResponseBase.IsClientConnected
+                        * to see if the client broke off the connection
+                        * and avoid trying to flush the response stream.
+                        * instead I'll swallow the exception that IIS throws in this situation
+                        * and rethrow anything else.*/
+                        if (httpException.Message
+                            == "An error occurred while communicating with the remote host. The error code is 0x80070057.") return;
+
+                        throw;
+                    }
+                }
+            }
+        }
+
+        public static string[] SplitOnFirst(this string strVal, string needle)
+        {
+            if (strVal == null)
+            {
+                return new string[0];
+            }
+
+            var pos = strVal.IndexOf(needle, StringComparison.OrdinalIgnoreCase);
+            return pos == -1
+                ? new[] { strVal }
+                : new[] { strVal.Substring(0, pos), strVal.Substring(pos + needle.Length) };
```

Comment:
Isn’t this pretty much the same as `strVal.Split(new [] { needle }, 2, StringSplitOptions.RemoveEmptyEntries)`?